x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/build_exceptions_query.ts

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Query } from '../../../../../../../../src/plugins/data/server';
+import { List, ListOperator, ListValues } from '../routes/schemas/types/lists_default_array';
+import { RuleAlertParams, Language } from '../types';
+
+interface Operators {
+  and: string;
+  not: string;
+  or: string;
+}
```

Comment:
Kind of a different way to express this. Normally I see people just use:
https://www.typescriptlang.org/docs/handbook/advanced-types.html#string-literal-types

like so:

```ts
type Operators = 'and' | 'not' | 'or';
```

And then skip the `keyof` like so:

```ts
export const getLanguageBooleanOperator = ({
  language,
  value,
}: {
  language: keyof Language;
  value: Operators;
}) => {
  if (language === 'lucene') {
    return value.toUpperCase();
  } else {
    return value;
  }
};
```

I would type the return type of this function with a `string` type. You could write a more verbose switch statement with stronger types between upper and lowercase if you really wanted to, but it's fine if you want to convert a set of string literal types to an upper case string. You do lose some typing though. A more through approach would be to do an exhaustive switch on the lower case types and return their upper case equivalents which is less dynamic for sure but more static types.

Reply:
I updated to try to make the types more static and easier for future developers to understand why we're capitalizing, etc. If this grows it should be easy to expand on.